talisker/sentry.py

```diff
@@ -159,7 +161,28 @@ def __call__(self, environ, start_response):
         # we clear the sentry context before the request starts, in order to
         # avoid picking up gunicorn log messages from previous requests
         self.client.context.clear()
-        return super().__call__(environ, start_response)
+        soft_start_timeout = talisker.get_config()['soft_request_timeout']
+        if soft_start_timeout >= 0:
+            # XXX get value of soft_start_timeout from config...how?
+            start_time = time.time()
+
+            def soft_timeout_start_response(status, headers, exc_info=None):
+                response = start_response(status, headers, exc_info=exc_info)
+                duration = (time.time() - start_time) * 1000
+                if (soft_start_timeout is not None and
+                        duration > soft_start_timeout):
+                    self.client.captureMessage(
+                        'Start_response over timeout: {}'
+                        .format(soft_start_timeout)
+                    )
+                # only clear transaction via url_rule if there is a flask
+                # context
+                if flask.has_request_context() and flask.request.url_rule:
+                    self.client.transaction.pop(flask.request.url_rule.rule)
```

Comment:
Rather than pop it here, I think I'd prefer to clear the transaction completely at the start of the request, after the self.client.context.clear() on line 163 above, do a self.client.transaction.clear(). Then we don't need to worry about a flask context, and we preserve the current requests transactions stack for as long is possible, incase any other issue occur.